Updated Date and Time Functions List

GabeFernandez310 opened the pull request:
### Description
Updates the functions list to include those newly supported for SQL queries.

**PDF of Printed Webpage:**
[Functions - OpenSearch documentation.pdf](https://github.com/Bit-Quill/documentation-website/files/11338361/Functions.-.OpenSearch.documentation.pdf)

### Issues Resolved

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

GabeFernandez310 committed:
Updated Date and Time Functions List

Signed-off-by: GabeFernandez310 <Gabriel.Fernandez@improving.com>

forestmvey commented on _search-plugins/sql/functions.md:
```diff
@@ -68,6 +68,7 @@ sinh | `sinh(number T) -> double` | `SELECT sinh(0.5) FROM my-index LIMIT 1`
 tan | `tan(number T) -> double` | `SELECT tan(0.5) FROM my-index LIMIT 1`
 
 ## Date and time
+Functions marked with * are not yet implemented in PPL, with plans to support them in the near future.
```
```suggestion
Functions marked with * have future PPL implementation on roadmap.
```

Yury-Fridlyand commented:
Missing: `addtime`, `convert_tz`, `curtime` and more. Please, compare list with `DateTimeFunction.java` or SQL plugin docs.

GabeFernandez310 committed:
Changed A Description

Signed-off-by: GabeFernandez310 <Gabriel.Fernandez@improving.com>

GabeFernandez310 committed:
Added More Function Descriptions

Signed-off-by: GabeFernandez310 <Gabriel.Fernandez@improving.com>

GabeFernandez310 committed:
Fixed An Entry

Signed-off-by: GabeFernandez310 <Gabriel.Fernandez@improving.com>

GabeFernandez310 commented:
> Missing: `addtime`, `convert_tz`, `curtime` and more. Please, compare list with `DateTimeFunction.java` or SQL plugin docs.

Fixed. Added all functions currently existing in `DateTimeFunction.java`
